core/src/main/java/org/akaza/openclinica/dao/hibernate/AuditUserLoginDao.java

```diff
@@ -38,7 +38,11 @@ public int getCountWithFilter(final AuditUserLoginFilter filter) {
     public ArrayList<AuditUserLoginBean> getWithFilterAndSort(final AuditUserLoginFilter filter, final AuditUserLoginSort sort, final int rowStart,
             final int rowEnd) {
         Criteria criteria = getCurrentSession().createCriteria(domainClass());
-        criteria = filter.execute(criteria);
```

Comment:
This is a good first effort but I'm not fully happy with it.

1. What is the root cause of the exception? It might be "LoginStatus.getByName" in the buildCriteria method (called by the execute method). So why is it thrown? Is there an invalid status name given?
2. The try catch block should include the rest of the code also.
3. I'm not a fan of silently failing. Isn't it an option to add the throws declaration for the IllegalArgumentException to the method and let the caller decide what to do with it?

Reply:
Good questions, sorry I'm just following up now.
1. The root cause here is because of an incorrect cast; criteria.list(), when empty, can't be cast to an ArrayList<>.  However, when there is data present in the list, this can cast without issue.
2. Agreed, it can be expanded to that, per my first attempt for a fix - https://github.com/tomhickerson/LibreClinica/commit/fa4375cc709e55459639372505636ec5bec1f2af
3. True, this could just contain 'throws' but then it's a larger fix as we have to decide in other classes, how to handle it.

Reply:
Going a little deeper into the root cause for 1:
- criteria.list() returns a List implementation when empty, which traditionally can't be cast to an ArrayList.  This is probably because a lot of the database code was written without Hibernate, and defaulted to returning ArrayLists for data [1]
- Hibernate Criteria was added later, and most of the classes were written to also return ArrayLists 

[1] - for example https://github.com/reliatec-gmbh/LibreClinica/blob/lc-develop/core/src/main/java/org/akaza/openclinica/dao/core/AuditableEntityDAO.java#L100